File: gcc/testsuite/gcc.dg/gomp/appendix-a/a.30.1.c
/* { dg-do compile } */

void
a30 (int n, float *a, float *b)
{
  int i;
#pragma omp parallel
  {
#pragma omp for lastprivate(i)
    for (i = 0; i < n - 1; i++)
      a[i] = b[i] + b[i + 1];
  }
  a[i] = b[i];			/* i == n-1 here */
}

File: gcc/testsuite/gfortran.dg/gomp/appendix-a/a.30.1.f90
! { dg-do compile }

      SUBROUTINE A30(N, A, B)
       INTEGER N
       REAL A(*), B(*)
        INTEGER I
!$OMP PARALLEL
!$OMP DO LASTPRIVATE(I)
        DO I=1,N-1
          A(I) = B(I) + B(I+1)
        ENDDO
!$OMP END PARALLEL
        A(I) = B(I)      ! I has the value of N here
      END SUBROUTINE A30
